src/shadertools/lib/shader-cache.js

```diff
@@ -68,4 +72,40 @@ export default class ShaderCache {
     }
     return shader;
   }
+
+  // Retrive Program from cache if exists, otherwise create new instance.
+  getProgram(gl, {vs, fs}) {
+    assert(typeof vs === 'string', ERR_SOURCE);
+    assert(typeof fs === 'string', ERR_SOURCE);
+    const cacheKey = `VertexShader: ${vs} FragmentShader: ${fs}`;
+
+    // Return if a matching Program exists in cache.
+    const program = this.programs[cacheKey];
+    if (program) {
+      assert(program.gl === gl, ERR_CONTEXT);
```

Comment:
I think a cache should be specific to a context, and we should only add programs that are specific to that context, if so we don't need a check here.